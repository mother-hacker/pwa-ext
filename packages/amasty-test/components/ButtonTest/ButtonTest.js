import React from 'react';
import { string } from 'prop-types';

const Button = props => {
    const {text} = props;

    return (
        <button>
            <span>{'Test button' + ' ' + text}</span>
        </button>
    );
};
Button.propTypes = {
    text: string
};

Button.defaultProps = {
    text: ''
};

export default Button;
