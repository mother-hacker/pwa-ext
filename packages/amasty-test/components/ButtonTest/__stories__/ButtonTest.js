import React from 'react';
import { storiesOf } from '@storybook/react';
import ButtonTest from '../ButtonTest';

const stories = storiesOf('Components/ButtonTest', module);

stories.add('Button test', () => (
    <ButtonTest
        text='Test some text'
    />
));
