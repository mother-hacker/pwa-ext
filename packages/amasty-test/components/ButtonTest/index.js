import ButtonTest from './ButtonTest';

export { ButtonTest }