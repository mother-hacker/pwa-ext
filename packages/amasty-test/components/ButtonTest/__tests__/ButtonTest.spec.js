import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import ButtonTest from '../ButtonTest';

const renderer = new ShallowRenderer();

test('Render button test with text property', () => {
    const component = renderer.render(<ButtonTest text='Some text' />);

    expect(component).toMatchSnapshot();
});

test('Render button test without text property', () => {
    const component = renderer.render(<ButtonTest />);

    expect(component).toMatchSnapshot();
});
