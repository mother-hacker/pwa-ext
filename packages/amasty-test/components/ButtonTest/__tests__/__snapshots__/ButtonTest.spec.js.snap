// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render button test with text property 1`] = `
<button>
  <span>
    Test button Some text
  </span>
</button>
`;

exports[`Render button test without text property 1`] = `
<button>
  <span>
    Test button 
  </span>
</button>
`;
